l3build-check.lua

```diff
@@ -55,7 +55,7 @@ local remove           = os.remove
 -- Set up the check system files: needed for checking one or more tests and
 -- for saving the test files
 function checkinit()
-  if not options["dirty"] then
+  if not Opts.dirty then
```

Comment:
While I would often be in favor of `a.b` instead of `a["b"]`, I would argue that for `options` in particular the `["..."]` should be kept even if the general convention is changed since quite often `["..."]` is required (because the key contains hyphens) and therefore switching between the syntax variants leads to inconsistent code.

Reply:
The actual code is already inconsistent on that respect (see `option_list` definition)
This change is just consistent in the inconsistency...